DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -298,5 +300,23 @@ int customisationId
                 new { customisationId, centreId, categoryId }
             ).FirstOrDefault();
         }
+
+        public IEnumerable<Course> GetCoursesAtCentreForCategoryId(int centreId, int categoryId)
+        {
+            return connection.Query<Course>(
+                @"SELECT
+                        c.CustomisationID,
+                        c.CentreID,
+                        c.ApplicationID,
+                        a.ApplicationName,
+                        c.CustomisationName,
+                        c.Active
+                    FROM Customisations AS c
+                    JOIN Applications AS a on a.ApplicationID = c.ApplicationID
+                    WHERE (CentreID = @centreId OR CentreID = 0)
+	                AND (a.CourseCategoryID = @categoryId OR @categoryId = 0)",

```

Comment:
I think I'd still prefer the categoryId to be null with no filter rather than "0" which can be ambiguous. What do you think?

Reply:
Admins don't have a nullable CategoryId. I think we should keep this behaviour consistent with the AdminUser records so it doesn't cause confusion unless we want to g through and change everything.

Reply:
But I think in this query the "categoryId" is effectively a filter on category id, which intuitively would be:
* Null = no filter
* Value = filter applied